system/lib/libc/lookup_name.c

```diff
@@ -0,0 +1,27 @@
+// Emscripten-specific version of musl/src/network/lookup_name.c
+
+#include <string.h>
+#include "musl/src/network/lookup.h"
+#include "emscripten_internal.h"
+
+int __lookup_name(struct address buf[static MAXADDRS], char canon[static 256], const char *name, int family, int flags)
+{
+	if (family != AF_INET || flags != AI_CANONNAME) {
+		return EAI_SYSTEM;
+	}
+
+	*canon = 0;
+	if (name) {
+		/* reject empty name and check len so it fits into temp bufs */
```

Comment:
I see a check for an empty name, but where is it rejected? (It looks like we continue down and send null to `UTF8ToString` in JS, which is fine.)

Reply:
I don't think we handle that case.. but also the old code didn't handle it either, so I'm temped to leave it as is.  I think passing NULL to gethostbyname is most likely undefined behaviour and I suppose we don't have a test for it.